pkg/kmm/containers_test.go

```diff
@@ -0,0 +1,230 @@
+
+		if len(testCase.args) > 0 {
+			assert.Equal(t, testCase.args, testBuilder.definition.Modprobe.Args.Load)
+		}
+
+		if len(testCase.rawargs) > 0 {
+			assert.Equal(t, testCase.rawargs, testBuilder.definition.Modprobe.RawArgs.Load)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithImagePullPolicy(t *testing.T) {
+	testCases := []struct {
+		imagePolicy    string
+		expectedPolicy string
+		expectedError  string
+	}{
+		{
+			imagePolicy:    "",
+			expectedPolicy: "IfNotPresent",
+			expectedError:  "'policy' can not be empty",
+		},
+		{
+			imagePolicy:    "Never",
+			expectedPolicy: "Never",
+			expectedError:  "",
+		},
+		{
+			imagePolicy:    "Always",
+			expectedPolicy: "Always",
+			expectedError:  "",
+		},
+		{
+			imagePolicy:    "IfNotPresent",
+			expectedPolicy: "IfNotPresent",
+			expectedError:  "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := NewModLoaderContainerBuilder("test")
+		testBuilder.WithImagePullPolicy(testCase.imagePolicy)
+
+		assert.Equal(t, testCase.expectedError, testBuilder.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, v1.ImagePullPolicy(testCase.expectedPolicy), testBuilder.definition.ImagePullPolicy)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithKernelMapping(t *testing.T) {
+	testCases := []struct {
+		mapping       *v1beta1.KernelMapping
+		expectedError string
+	}{
+		{
+			mapping:       buildRegExKernelMapping(""),
+			expectedError: "'mapping' can not be empty nil",
+		},
+		{
+			mapping:       buildRegExKernelMapping("^.+$"),
+			expectedError: "",
+		},
+		{
+			mapping:       buildLiteralKernelMapping("5.14.0-70.58.1.el9_0.x86_64"),
+			expectedError: "",
+		},
+		{
+			mapping:       buildLiteralKernelMapping(""),
+			expectedError: "'mapping' can not be empty nil",
+		},
+	}
+
+	for _, testcase := range testCases {
+		testBuilder := NewModLoaderContainerBuilder("test")
+		testBuilder.WithKernelMapping(testcase.mapping)
+
+		if testcase.expectedError != "" {
+			assert.Equal(t, testBuilder.errorMsg, testcase.expectedError)
+		} else {
+			assert.Equal(t, testBuilder.definition.KernelMappings[0], *testcase.mapping)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithOptions(t *testing.T) {
+
+}
+
+func TestModuleLoaderContainerBuilder_WithVersion(t *testing.T) {
+	testCases := []struct {
+		version       string
+		expectedError string
+	}{
+		{
+			version:       "",
+			expectedError: "'version' can not be empty",
+		},
+		{
+			version:       "1.1",
+			expectedError: "",
+		},
+	}
+
+	for _, testcase := range testCases {
+		testBuilder := NewModLoaderContainerBuilder("test")
+		testBuilder.WithVersion(testcase.version)
+
+		if testcase.expectedError != "" {
+			assert.Equal(t, testBuilder.errorMsg, testcase.expectedError)
+		} else {
+			assert.Equal(t, testBuilder.definition.Version, testcase.version)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_BuildModuleLoaderContainerCfg(t *testing.T) {
+
```

Comment:
In this function you can 
1.build valid ModLoaderContainerBuilder and check that error is nil + ModuleLoaderContainerSpec is correct
 2. build invalid ModLoaderContainerBuilder and check that error is NOT nil + ModuleLoaderContainerSpec is nil
 3. build valid ModLoaderContainerBuilder but mutate it with invalid value in With function and check that error is Not nil + ModuleLoaderContainerSpec is nil